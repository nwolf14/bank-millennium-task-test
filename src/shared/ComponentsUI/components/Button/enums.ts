export enum Color {
  primary = 'primary',
  secondary = 'secondary',
  success = 'success',
  warning = 'warning',
  error = 'error',
}

export enum Size {
  small = 'small',
  medium = 'medium',
  large = 'large',
}

export enum Variant {
  text = 'text',
  contained = 'contained',
  outlined = 'outlined',
}
