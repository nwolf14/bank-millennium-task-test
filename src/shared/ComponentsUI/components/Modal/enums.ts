export enum ModalId {
  storyBookModal = 'stroryBookModal',
}
