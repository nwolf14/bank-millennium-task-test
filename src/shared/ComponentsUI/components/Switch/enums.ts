export enum Size {
  small = 'small',
  medium = 'medium',
}
