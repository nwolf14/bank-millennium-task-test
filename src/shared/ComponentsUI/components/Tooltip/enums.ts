export enum TooltipPosition {
  topStart = 'top-start',
  topCenter = 'top-center',
  topEnd = 'top-end',
  bottomStart = 'bottom-start',
  bottomCenter = 'bottom-center',
  bottomEnd = 'bottom-end',
  leftStart = 'left-start',
  leftCenter = 'left-center',
  leftEnd = 'left-end',
  rightStart = 'right-start',
  rightCenter = 'right-center',
  rightEnd = 'right-end',
}
