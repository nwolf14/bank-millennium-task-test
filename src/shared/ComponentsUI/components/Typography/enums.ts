export enum Variant {
  h1 = 'h1',
  h2 = 'h2',
  h3 = 'h3',
  h4 = 'h4',
  h5 = 'h5',
  h6 = 'h6',
  text = 'p',
  smallText = 'small',
}

export enum FontStyle {
  italic = 'italic',
  normal = 'normal',
}

export enum FontWeight {
  black = 'black',
  bold = 'bold',
  extraBold = 'extra-bold',
  light = 'light',
  medium = 'medium',
  regular = 'regular',
  semiBold = 'semi-bold',
}
