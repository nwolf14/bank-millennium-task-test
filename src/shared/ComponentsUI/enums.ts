export enum Margin {
  marginTop = 'margin-top',
  marginBottom = 'margin-bottom',
  marginTopBottom = 'margin-top-bottom',
  marginNone = 'margin-none',
}
