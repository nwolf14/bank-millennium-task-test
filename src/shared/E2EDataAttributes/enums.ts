export enum E2EAttribute {
  componentBlock = 'component-block',
  componentBlockNode = 'component-block-node',
  componentConnector = 'component-connector',
  componentConnectorPath = 'component-line-path',
  componentsPanelBlock = 'components-panel-block',
  editBlockCoordinatesInput = 'edit-block-coodinates-input',
  editConditionBlockInputRadio = 'edit-condition-block-input-radio',
  editConditionBlockInputText = 'edit-condition-block-input-text',
  editConditionBlockSelect = 'edit-condition-block-select',
  editConnectorInputRadio = 'edit-connector-input-radio',
  navigationLayer = 'navigation-layer',
  text = 'text',
  unselected = 'unselected',
}
