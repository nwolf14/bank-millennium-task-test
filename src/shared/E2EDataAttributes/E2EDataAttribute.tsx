import React, { Children, FC, ReactChild, ReactElement } from 'react';

// others
import { DATA_TEST_PREFIX } from './constants';
import { E2EAttribute } from './enums';

type TProps = {
  children: ReactElement<ReactChild>;
  type?: E2EAttribute;
  value?: number | string;
};

const E2EDataAttribute: FC<TProps> = ({ children, type = '', value = '' }) => {
  const childrenOnly = Children.only(children);

  if (!type && !value) {
    return childrenOnly;
  }

  return React.cloneElement(childrenOnly, {
    [`${DATA_TEST_PREFIX}-${type}`]: value,
  });
};

export default E2EDataAttribute;
