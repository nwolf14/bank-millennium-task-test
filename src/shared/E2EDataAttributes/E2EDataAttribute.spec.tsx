import { render } from '@testing-library/react';

// components
import E2EDataAttribute from './E2EDataAttribute';

// others
import { E2EAttribute } from './enums';

describe('E2EDataAttribute', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <E2EDataAttribute type={E2EAttribute.componentBlock} value={1}>
        <span>child</span>
      </E2EDataAttribute>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('E2EDataAttribute', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <E2EDataAttribute>
        <span>child</span>
      </E2EDataAttribute>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
