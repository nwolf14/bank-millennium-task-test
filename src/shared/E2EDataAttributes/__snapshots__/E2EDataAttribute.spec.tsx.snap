// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2EDataAttribute matches snapshot 1`] = `
<DocumentFragment>
  <span
    data-test-component-block="1"
  >
    child
  </span>
</DocumentFragment>
`;

exports[`E2EDataAttribute matches snapshot 2`] = `
<DocumentFragment>
  <span>
    child
  </span>
</DocumentFragment>
`;
